plugins/channel_header_plug/channel_header_plug.tsx

```diff
@@ -123,9 +127,20 @@ class ChannelHeaderPlug extends React.PureComponent<ChannelHeaderPlugProps, Chan
         super(props);
         this.state = {
             dropdownOpen: false,
+            disableButtonsClosingRHS: false,
         };
     }
 
+    componentDidUpdate(prevProps: ChannelHeaderPlugProps) {
+        if (prevProps.sidebarOpen && !this.props.sidebarOpen) {
+            this.setState({disableButtonsClosingRHS: true});
+
+            setTimeout(() => {
+                this.setState({disableButtonsClosingRHS: false});
+            }, CHANNEL_HEADER_PLUG_DISABLE_TIMEOUT);
+        }
+    }
```

Comment:
I tried to implement this with `getDerivedStateFromProps`, but since we need to perform a side effect, it couldn't be done with a static method.